add option to place toggle action to the left of title

noodlemoodle opened the pull request:
TODO test

noodlemoodle committed:
add option to place toggle action to the left of title

KernelDeimos commented on src/foam/u2/borders/CollapseBorder.js:
```diff
@@ -77,15 +82,18 @@ foam.CLASS({
         addClass(this.myClass()).
         enableClass('expanded', this.expanded$).
         start('div').
+          style({ 'display': 'flex', 'flex-direction': 'row', 'align-items': 'center' }).
```
This should be moved to a CSS axiom. Looks great though!

sarthak-marwaha commented on src/foam/u2/borders/CollapseBorder.js:
```diff
@@ -77,15 +82,18 @@ foam.CLASS({
           addClass(this.myClass('toolbar')).
+          start('div').
+            addClass(this.myClass('control')).
+            addClass( this.toggleLeft ? '' : this.myClass('float-right')).
```
Could use `.enableClass(this.myClass('float-right'), this.toggleLeft$)` here

sarthak-marwaha commented on src/foam/u2/borders/CollapseBorder.js:
```diff
@@ -77,15 +82,18 @@ foam.CLASS({
+            tag(this.TOGGLE, { label: this.label$ }).
+          end().
           start('span').
+            on('click', () => this.toggle()).
```
Why are we replacing the button with a listener? This will cause accessibility issues

noodlemoodle replied:
im using this for a part of the transaction plan cards, would be a nice to have if this could toggle on click of the title as well, since the button is a bit small on mobile

noodlemoodle committed:
fix no floating in flex

noodlemoodle committed:
left a comment todo

noodlemoodle commented:
addressed all in https://github.com/kgrgreer/foam3/pull/557/commits/3be51c1e55165f505d8388ff1e5959ca6d81832e @KernelDeimos @sarthak-marwaha 

removed click listener for title for now andd left a comment todo, will find a better way to do this

noodlemoodle committed:
null check data

noodlemoodle committed:
oops made fix on the wrong branch
